packages/core/admin/webpack.config.js

```diff
@@ -128,41 +114,83 @@ module.exports = ({
       rules: [
         {
           test: /\.m?js$/,
-          // TODO remove when plugins are built separately
-          include: [cacheDir, ...pluginsPath],
+          include: cacheDir,
+          oneOf: [
+            // Use babel-loader for files that distinct the ee and ce code
+            // These files have an import Something from 'ee_else_ce/
+            {
+              test(filePath) {
+                if (!filePath) {
+                  return false;
+                }
+
+                try {
+                  const fileContent = fse.readFileSync(filePath).toString();
+
+                  if (fileContent.match(/from.* ['"]ee_else_ce\//)) {
+                    return true;
+                  }
+
+                  return false;
+                } catch (e) {
+                  return false;
+                }
+              },
+              use: {
+                loader: require.resolve('babel-loader'),
+                options: {
+                  cacheDirectory: true,
+                  cacheCompression: isProduction,
+                  compact: isProduction,
+                  presets: [
+                    require.resolve('@babel/preset-env'),
+                    require.resolve('@babel/preset-react'),
+                  ],
+                  plugins: [
+                    [
+                      require.resolve('@strapi/babel-plugin-switch-ee-ce'),
+                      {
+                        // Imported this tells the custom plugin where to look for the ee folder
+                        roots,
+                      },
+                    ],
+                    require.resolve('@babel/plugin-proposal-class-properties'),
+                    require.resolve('@babel/plugin-syntax-dynamic-import'),
+                    require.resolve('@babel/plugin-transform-modules-commonjs'),
+                    require.resolve('@babel/plugin-proposal-async-generator-functions'),
+
+                    [
+                      require.resolve('@babel/plugin-transform-runtime'),
```

Comment:
I think all except `@babel/plugin-transform-runtime` are now part of `@babel/preset-env`, so we don't have to include them manually anymore.

Reply:
You might be right I need to check and also in order to make the esbuild-loader working I add to add a babel/polyfill in the entry point of the webpack config, before compiling the admin files so it might not be needed at all you are right.

Reply:
In fact it's not the same usage this plugin `enables the re-use of Babel's injected helper code to save on codesize` but since we are only using babel-loader for the files that have the `ee_else_ce` import I guess having this plugin or not does not make a huge impact on the code size so let's remove it.